src/webview/common-ext/createComponentHelpers.ts

```diff
@@ -251,25 +253,42 @@ export function getDevfileRegistries(): DevfileRegistry[] {
  * @returns a list of the devfile capabilities
  */
 export function getDevfileCapabilities(): string[] {
-    return [ 'Debug Support', 'Deploy Support'];
+    return ['Debug Support', 'Deploy Support'];
 }
 
 /**
  * Returns a list of the devfile tags found in devfiles registries.
  *
  * @returns a list of the devfile tags
  */
-export function getDevfileTags(url?:string ): string[] {
+export function getDevfileTags(url?: string): string[] {
     const devfileRegistries = getDevfileRegistries();
 
     const devfileTags: string[] = [
-            ...new Set(
-                devfileRegistries
-                    .filter((devfileRegistry) => url ? devfileRegistry.url === url : true)
-                    .flatMap((_devfileRegistry) => _devfileRegistry.devfiles)
-                    .flatMap((_devfile) => _devfile.tags))
-        ]
-        .sort((a, b) => a.localeCompare(b));
+        ...new Set(
+            devfileRegistries
+                .filter((devfileRegistry) => url ? devfileRegistry.url === url : true)
+                .flatMap((_devfileRegistry) => _devfileRegistry.devfiles).sort(devfileSort)
+                .flatMap((_devfile) => _devfile.tags))
+    ]
+    return devfileTags.filter((devfileTag) => !devfileTag.toLowerCase().includes('deprecate'));
+}
+
+function devfileSort(a: Devfile, b: Devfile): number {
+    const QUARKUS_REGEX = /[Qq]uarkus/;
+    const aQuarkus = QUARKUS_REGEX.test(a.name);
+        const bQuarkus = QUARKUS_REGEX.test(b.name);
+        if (aQuarkus && !bQuarkus) {
+            return -1;
+        } else if (bQuarkus && !aQuarkus) {
+            return 1;
+        }
```

Comment:
I don't understand why are the `Quarkus`  Devfile(-s) are explicitly processed here... 
If we want `Quarkus` Devfiles to be always "promoted" to be on top - it's OK, but doesn't work, because selection of a tag and following deselection makes any Tag (including the `Quarkus` one) to disappear from the visible part of the list.
[Screencast from 2023-12-04 13-45-07.webm](https://github.com/redhat-developer/vscode-openshift-tools/assets/620781/78adc447-fea9-40cc-8d85-66bfb750ad8d)